File: src/types/achievement.ts
// src/types/achievement.ts

export type AchievementFrameType = 'task' | 'goal' | 'challenge' | string;

export interface AdvancementItemDto {
  advancementId: string;
  namespace?: string;
  parentId?: string | null;
  title: string;
  description: string | null;
  iconRelPath: string;
  frameType: AchievementFrameType;
  isCompleted: boolean;
  unlockedAt: number | null;
  isFirstCareerUnlock: boolean;
  criteriaData?: Record<string, string> | null;

  // 兼容字段别名 (snake_case)
  advancement_id?: string;
  frame_type?: AchievementFrameType;
  is_completed?: boolean;
  unlocked_at?: number | null;
  is_first_career_unlock?: boolean;
  criteria_json?: string | null;
}

export interface GameSessionDto {
  id: string;
  instanceId: string;
  worldName: string | null;
  playerUuid: string;
  playerName: string | null;
  startedAt: number;
  endedAt: number;
  durationSecs: number;
  exitCode: number | null;
  newAdvancementsCount: number;

  // 兼容字段别名 (snake_case)
  instance_id?: string;
  world_name?: string | null;
  player_uuid?: string;
  player_name?: string | null;
  started_at?: number;
  ended_at?: number;
  duration_secs?: number;
  exit_code?: number | null;
  new_advancements_count?: number;
}

export interface CareerSummaryDto {
  playerUuid: string;
  totalPlayTimeSecs: number;
  totalSessions: number;
  totalCareerAdvancements: number;
  instancesPlayed: number;
  firstPlayedAt: number | null;
  lastPlayedAt: number | null;

  // 兼容字段别名 (snake_case)
  player_uuid?: string;
  total_play_time_secs?: number;
  total_sessions?: number;
  total_career_advancements?: number;
  instances_played?: number;
  first_played_at?: number | null;
  last_played_at?: number | null;
}

File: src-tauri/src/domain/achievement.rs
// src-tauri/src/domain/achievement.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GameSession {
    pub id: String,
    pub instance_id: String,
    pub world_name: Option<String>,
    pub player_uuid: String,
    pub player_name: Option<String>,
    pub started_at: i64,
    pub ended_at: i64,
    pub duration_secs: i64,
    pub exit_code: i32,
    pub new_advancements_count: i64,
    pub summary_json: Option<String>,
    pub created_at: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct InstanceAdvancement {
    pub id: String,
    pub instance_id: String,
    pub world_name: String,
    pub player_uuid: String,
    pub advancement_id: String,
    pub frame_type: String,
    pub session_id: Option<String>,
    pub unlocked_at: i64,
    pub is_first_career_unlock: bool,
    pub criteria_json: Option<String>,
    pub created_at: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AdvancementMetadataCache {
    pub advancement_id: String,
    pub namespace: String,
    pub parent_id: Option<String>,
    pub title: String,
    pub description: Option<String>,
    pub frame_type: String,
    pub icon_rel_path: String,
    pub source_type: String,
    pub source_hash: Option<String>,
    pub updated_at: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AdvancementItemDto {
    pub advancement_id: String,
    pub namespace: String,
    pub parent_id: Option<String>,
    pub title: String,
    pub description: Option<String>,
    pub icon_rel_path: String,
    pub frame_type: String,
    pub unlocked_at: Option<i64>,
    pub is_completed: bool,
    pub is_first_career_unlock: bool,
    pub criteria_data: Option<serde_json::Value>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GameSessionDto {
    pub id: String,
    pub instance_id: String,
    pub world_name: Option<String>,
    pub player_uuid: String,
    pub player_name: Option<String>,
    pub started_at: i64,
    pub ended_at: i64,
    pub duration_secs: i64,
    pub exit_code: i32,
    pub new_advancements_count: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GameSessionDetailDto {
    pub session: GameSessionDto,
    pub new_advancements: Vec<AdvancementItemDto>,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct CareerSummaryDto {
    pub total_playtime_secs: i64,
    pub total_sessions_count: i64,
    pub total_unique_advancements: i64,
    pub challenge_advancements_count: i64,
    pub last_played_at: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AchievementUnlockedEventPayload {
    pub instance_id: String,
    pub world_name: String,
    pub player_uuid: String,
    pub player_name: Option<String>,
    pub advancement_id: String,
    pub title: String,
    pub description: Option<String>,
    pub icon_rel_path: String,
    pub frame_type: String,
    pub unlocked_at: i64,
    pub is_first_career_unlock: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AchievementSessionSummaryPayload {
    pub session_id: String,
    pub instance_id: String,
    pub world_name: Option<String>,
    pub player_uuid: String,
    pub duration_secs: i64,
    pub new_advancements_count: i64,
    pub new_advancements: Vec<AchievementUnlockedEventPayload>,
}
